lib/measures/urban_geometry_creation/measure.rb

```diff
@@ -713,46 +89,41 @@ def run(model, runner, user_arguments)
       space = stories[i].spaces.first
       if space && space.spaceType.is_initialized
         space_type = space.spaceType.get
-      else  
+      else
         space_type = OpenStudio::Model::SpaceType.new(model)
         runner.registerInfo("Story #{i} does not have a space type, creating new one")
       end
       space_types[i] = space_type
     end
-    
+
     # delete the previous building
     model.getBuilding.remove
-    
+
     # create new building and transfer default construction set
     model.getBuilding.setDefaultConstructionSet(default_construction_set)
-    
+
     # instance variables
     @runner = runner
     @origin_lat_lon = nil
-    @geojson = nil
-    
+
     path = @runner.workflow.findFile(geojson_file)
     if path.nil? || path.empty?
       @runner.registerError("GeoJSON file '#{geojson_file}' could not be found")
       return false
     end
-    
+
     path = path.get.to_s
     if !File.exists?(path)
       @runner.registerError("GeoJSON file '#{path}' could not be found")
       return false
     end
 
-    File.open(path, 'r') do |file|
-      @geojson = JSON.parse(file.read, {symbolize_names: true})
-    end
-
-    feature = get_feature(feature_id)
+    feature = geojson_gem.get_feature(feature_id, path)
```

Comment:
I think there should be a class constructed from the path that reads in the GeoJSON once.  You would create an instance of that class from the path, then later call methods like `get_feature` on it.  We might need to rename some classes to make better sense:

URBANopt::GeoJSON::GeoJSON -> URBANopt::GeoJSON::Extension
Class to read a GeoJSON file -> URBANopt::GeoJSON::GeoJSON or URBANopt::GeoJSON::GeoJSONFile

See what we did in this repo (very similar just reading a different type of file)
https://github.com/ladybug-tools-in2/energy-model-measure/tree/develop/lib/ladybug